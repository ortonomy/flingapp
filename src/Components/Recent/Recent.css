.Recent .recent-container {
  padding: 1em 0;
}

.Recent .recent-container > i {
  color: #ffde17;
  font-size: 16px;
}

.Recent .recent-container span {
  text-transform: uppercase;
  padding-left: 0.5em;
}
