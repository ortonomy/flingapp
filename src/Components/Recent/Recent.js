import React, { Component } from 'react';
import RecentItem from '../RecentItem/RecentItem';
import './Recent.css';

const recentFreelancers = [{
  name: 'Jorge Brock',
  url: 'freelancer/101'
},{
  name: 'Lettie Hicks',
  url: 'freelancer/102'
},{
  name: 'Winifred Summers',
  url: 'reelancer/103'
}]


const recentProjects = [{
  name: 'DLA General English',
  url: 'project/101'
},{
  name: 'BE Update 2017',
  url: 'project/102'
},{
  name: 'Wordlists and Images',
  url: 'project/103'
}]

class Recent extends Component {
  render() {
    const projects = []
    const freelancers = []

    for (let f of recentFreelancers) {
      freelancers.push(<RecentItem icon='user' displayText={f.name} linkTarget={f.url} />)
    }

    for (let p of recentProjects) {
      projects.push(<RecentItem icon='folder-o' displayText={p.name} linkTarget={p.url} />)
    }

    return (
      <div className="Recent">
        <div className='sidebar-section-title'>
          <i className='fa fa-eye'></i>
          <span>Recently Viewed</span>
        </div>
        <div className='recent-container'>
          <i className="fa fa-users" aria-hidden="true"></i>
          <span>Freelancers</span>
          {freelancers}
        </div>
        <div className='recent-container'>
          <i className="fa fa-sticky-note" aria-hidden="true"></i>
          <span>Projects</span>
          {projects}
        </div>
      </div>
    )
  }
}

export default Recent;
