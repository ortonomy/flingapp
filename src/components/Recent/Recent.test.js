import React from 'react';
import Recent from '../Recent';


it('renders without crashing', () => {
  const wrapper = shallow(<Recent />);
  expect(wrapper).toMatchSnapshot();
});
