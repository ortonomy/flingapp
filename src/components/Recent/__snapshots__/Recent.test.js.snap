// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Recent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <i
                    className="fa fa-eye"
          />
          <span>
                    Recently Viewed
          </span>
</div>,
        <div
          className={undefined}
>
          <i
                    aria-hidden="true"
                    className="fa fa-users"
          />
          <span>
                    Freelancers
          </span>
          <RecentItem
                    displayText="Jorge Brock"
                    icon="user"
                    linkTarget="freelancer/101"
          />
          <RecentItem
                    displayText="Lettie Hicks"
                    icon="user"
                    linkTarget="freelancer/102"
          />
          <RecentItem
                    displayText="Winifred Summers"
                    icon="user"
                    linkTarget="reelancer/103"
          />
</div>,
        <div
          className={undefined}
>
          <i
                    aria-hidden="true"
                    className="fa fa-sticky-note"
          />
          <span>
                    Projects
          </span>
          <RecentItem
                    displayText="DLA General English"
                    icon="folder-o"
                    linkTarget="project/101"
          />
          <RecentItem
                    displayText="BE Update 2017"
                    icon="folder-o"
                    linkTarget="project/102"
          />
          <RecentItem
                    displayText="Wordlists and Images"
                    icon="folder-o"
                    linkTarget="project/103"
          />
</div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-eye"
/>,
            <span>
              Recently Viewed
</span>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-eye",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Recently Viewed",
            },
            "ref": null,
            "rendered": "Recently Viewed",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              aria-hidden="true"
              className="fa fa-users"
/>,
            <span>
              Freelancers
</span>,
            Array [
              <RecentItem
                displayText="Jorge Brock"
                icon="user"
                linkTarget="freelancer/101"
/>,
              <RecentItem
                displayText="Lettie Hicks"
                icon="user"
                linkTarget="freelancer/102"
/>,
              <RecentItem
                displayText="Winifred Summers"
                icon="user"
                linkTarget="reelancer/103"
/>,
            ],
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-users",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Freelancers",
            },
            "ref": null,
            "rendered": "Freelancers",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "Jorge Brock",
            "nodeType": "function",
            "props": Object {
              "displayText": "Jorge Brock",
              "icon": "user",
              "linkTarget": "freelancer/101",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Lettie Hicks",
            "nodeType": "function",
            "props": Object {
              "displayText": "Lettie Hicks",
              "icon": "user",
              "linkTarget": "freelancer/102",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Winifred Summers",
            "nodeType": "function",
            "props": Object {
              "displayText": "Winifred Summers",
              "icon": "user",
              "linkTarget": "reelancer/103",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              aria-hidden="true"
              className="fa fa-sticky-note"
/>,
            <span>
              Projects
</span>,
            Array [
              <RecentItem
                displayText="DLA General English"
                icon="folder-o"
                linkTarget="project/101"
/>,
              <RecentItem
                displayText="BE Update 2017"
                icon="folder-o"
                linkTarget="project/102"
/>,
              <RecentItem
                displayText="Wordlists and Images"
                icon="folder-o"
                linkTarget="project/103"
/>,
            ],
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-sticky-note",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Projects",
            },
            "ref": null,
            "rendered": "Projects",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "DLA General English",
            "nodeType": "function",
            "props": Object {
              "displayText": "DLA General English",
              "icon": "folder-o",
              "linkTarget": "project/101",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "BE Update 2017",
            "nodeType": "function",
            "props": Object {
              "displayText": "BE Update 2017",
              "icon": "folder-o",
              "linkTarget": "project/102",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Wordlists and Images",
            "nodeType": "function",
            "props": Object {
              "displayText": "Wordlists and Images",
              "icon": "folder-o",
              "linkTarget": "project/103",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <i
                        className="fa fa-eye"
            />
            <span>
                        Recently Viewed
            </span>
</div>,
          <div
            className={undefined}
>
            <i
                        aria-hidden="true"
                        className="fa fa-users"
            />
            <span>
                        Freelancers
            </span>
            <RecentItem
                        displayText="Jorge Brock"
                        icon="user"
                        linkTarget="freelancer/101"
            />
            <RecentItem
                        displayText="Lettie Hicks"
                        icon="user"
                        linkTarget="freelancer/102"
            />
            <RecentItem
                        displayText="Winifred Summers"
                        icon="user"
                        linkTarget="reelancer/103"
            />
</div>,
          <div
            className={undefined}
>
            <i
                        aria-hidden="true"
                        className="fa fa-sticky-note"
            />
            <span>
                        Projects
            </span>
            <RecentItem
                        displayText="DLA General English"
                        icon="folder-o"
                        linkTarget="project/101"
            />
            <RecentItem
                        displayText="BE Update 2017"
                        icon="folder-o"
                        linkTarget="project/102"
            />
            <RecentItem
                        displayText="Wordlists and Images"
                        icon="folder-o"
                        linkTarget="project/103"
            />
</div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-eye"
/>,
              <span>
                Recently Viewed
</span>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-eye",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recently Viewed",
              },
              "ref": null,
              "rendered": "Recently Viewed",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                className="fa fa-users"
/>,
              <span>
                Freelancers
</span>,
              Array [
                <RecentItem
                  displayText="Jorge Brock"
                  icon="user"
                  linkTarget="freelancer/101"
/>,
                <RecentItem
                  displayText="Lettie Hicks"
                  icon="user"
                  linkTarget="freelancer/102"
/>,
                <RecentItem
                  displayText="Winifred Summers"
                  icon="user"
                  linkTarget="reelancer/103"
/>,
              ],
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-users",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Freelancers",
              },
              "ref": null,
              "rendered": "Freelancers",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "Jorge Brock",
              "nodeType": "function",
              "props": Object {
                "displayText": "Jorge Brock",
                "icon": "user",
                "linkTarget": "freelancer/101",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Lettie Hicks",
              "nodeType": "function",
              "props": Object {
                "displayText": "Lettie Hicks",
                "icon": "user",
                "linkTarget": "freelancer/102",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Winifred Summers",
              "nodeType": "function",
              "props": Object {
                "displayText": "Winifred Summers",
                "icon": "user",
                "linkTarget": "reelancer/103",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                className="fa fa-sticky-note"
/>,
              <span>
                Projects
</span>,
              Array [
                <RecentItem
                  displayText="DLA General English"
                  icon="folder-o"
                  linkTarget="project/101"
/>,
                <RecentItem
                  displayText="BE Update 2017"
                  icon="folder-o"
                  linkTarget="project/102"
/>,
                <RecentItem
                  displayText="Wordlists and Images"
                  icon="folder-o"
                  linkTarget="project/103"
/>,
              ],
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-sticky-note",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Projects",
              },
              "ref": null,
              "rendered": "Projects",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "DLA General English",
              "nodeType": "function",
              "props": Object {
                "displayText": "DLA General English",
                "icon": "folder-o",
                "linkTarget": "project/101",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "BE Update 2017",
              "nodeType": "function",
              "props": Object {
                "displayText": "BE Update 2017",
                "icon": "folder-o",
                "linkTarget": "project/102",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Wordlists and Images",
              "nodeType": "function",
              "props": Object {
                "displayText": "Wordlists and Images",
                "icon": "folder-o",
                "linkTarget": "project/103",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
